README.markdown

```diff
@@ -365,34 +455,76 @@ The example above demonstrates the following style guidelines:
  + Don't add modifiers such as `internal` when they're already the default. Similarly, don't repeat the access modifier when overriding a method.
  + Organize extra functionality (e.g. printing) in extensions.
  + Hide non-shared, implementation details such as `centerString` inside the extension using `private` access control.
+ + Take advantage of Swift 5.1's implicit `return` in one line closures.
 
 ### Use of Self
 
 For conciseness, avoid using `self` since Swift does not require it to access an object's properties or invoke its methods.
 
-Use self only when required by the compiler (in `@escaping` closures, or in initializers to disambiguate properties from arguments). In other words, if it compiles without `self` then omit it.
-
+Use `self` only when required by the compiler (e.g. in `@escaping` closures) or inside `init`'s when initializing each property (i.e. not just to disambiguate properties from arguments, but on **all** of them).
 
 ### Computed Properties
 
 For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided.
 
-**Preferred:**
+If possible, take advantage of Swift 5.1's implicit `return` in one line closures, and put the closure body inline with the property declaration.
+
+**Preferred**:
 ```swift
-var diameter: Double {
-  return radius * 2
-}
+var diameter: Double { radius * 2 }
 ```
 
-**Not Preferred:**
+**Not Preferred**:
 ```swift
 var diameter: Double {
-  get {
-    return radius * 2
-  }
+    get {
+      return radius * 2
+    }
 }
 ```
 
+### Shadowing private members
+
+Sometimes we need to expose some members like properties and functions publicly (e.g. for read-only property), but have a **shadowing** version for private usage (e.g. read-write property). 
+
+* When shadowing a property, the private property's name **should** be prefixed with a `_`, instead of more verbose alternatives like  `mutatingX`, etc..
+
+* When shadowing a function, the private function's name **can** be prefixed with a `_`, in the lack of a more well suited name to define it.
+
+**Preferred**:
+```swift
+var state: Int { _state }
+private var _state: Int
+
+let dynamicState: Property<Int>
+private let _dynamicState: MutableProperty<Int>
+
+func reticulateSplines(spline: [String])
```

Comment:
Accordingly with `#Prose` section, shouldn't it be better to name this method `reticulate(splines: [String])`. The same applies to the private method bellow.

Reply:
I think you mean more the `#Naming` section, perhaps?

Following Swift [API Design guidelines](https://swift.org/documentation/api-design-guidelines/) Argument Labels section below, I agree that we should change the signatures, but maybe we should omit the label since it's a part of a grammatical phrase:

![Screenshot 2019-10-24 at 12 16 49](https://user-images.githubusercontent.com/1391324/67480798-79a62f80-f658-11e9-98ec-639c0f6ba03c.png)

So, perhaps we should use `func reticulateSplines(_ splines: [String])` instead? 🤔

Reply:
honestly, i think it's a decision for the team, i prefer `func reticulate(splines: ...` but `func reticulateSplines(_ splines: ...` it's more readable.